### ISSUE TITLE
Proposal for a MDP repository

### ISSUE THREAD

**Issue by pslacerda** [Becksteinlab/GromacsWrapper#81]
What do you think about the idea of create a repository of .mdp files? It may be very helpful.

Maybe not the ones we are using right now. I don't have many in hand as I'm not doing many simulations these days (mostly a programming guy). The .mdp options keep changing and some old files I have are already somewhat outdated.

There isn't so many kinds of .mdp out there. We may create a site where users can adjust the relevant options and simply copy & paste or download into their projects. Power users will probably not need it, except if they are lazy (which many are), but many Gromacs users may find it helpful to not keep a personal .mdp repository themselves or recreate the files every time. Not speaking about users that know what to do but have no idea on how to do.

When the user know what he is doing is more a matter of join together mdp options and adjust parameters. Maybe we can help people at doing this. Molecular dynamics for noobs and repository for the rest, along with previously made GromacsWrapper programs we can seriously help to popularize molecular simulations.

It can be as simple as one or two interactive webpages. The main expertise here is the .mdp excerpt templates used for build a complete .mdp file. The user select which excerpts to use than adjust its options manually if not satisfied with the provided defaults.

**Comment by orbeckst** [comment]
Creating a repository always comes with responsibility. In some ways, shipping MDPs with GW is already a bit of a burden but given it's hackish nature (and the convenience for me...) made it ok. Building good repositories is not that easy – we created [Lipidbook](http://lipidbook.bioch.ox.ac.uk/) and [Ligandbook](http://ligandbook.org/) and that was hard work to get right. [CHARMM-GUI](http://www.charmm-gui.org/) ships Gromacs input files and I know that people are using them directly.

I use the stuff in `gromacs.setup` but I honestly don't know who else and that's the only part where the MDP files are used in GW.

And truth be told – if my students told me they were doing simulations with MDP files they found on the internet I would probably tell them to `rm -rf`, read the appropriate papers, and then create MDP files with the appropriate settings. (Actually, more likely I'd tell them to read the papers and to use one of the MDP files that we keep internally but these are still the settings that we as a lab decided to use and other people might disagree.) 

Basically, I don't want to shoulder all the responsibility for "correct" input files. (And yes, `gromacs.setup` does go in this direction but at least the MDP files have a big, fat disclaimer...).

Bottomline: If you can come up with a workable solution that won't allow people to blame bad simulations on us then I am happy to listen to ideas. I wouldn't make it part of GW per se, though – if anything, rather make the package slimmer (along the discussion #82 and name space packages). Perhaps the idea of an input generator would be a good compromise: still requires user input but you can give some a bit of a helping hand and catch some common mistakes in advance (e.g., the plethora of cutoffs for non-bonded interactions...).

**Comment by orbeckst** [comment]
I am going to close this issue as there hasn't been any recent development. We can re-open if there's enough interest.
